Gemfile

```
@@ -3,6 +3,20 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
 ruby "3.1.2"
 
+gem 'rails-i18n'
+
+gem "bcrypt"
+
+gem 'whenever', require: false
+
+gem "bootstrap_form"
+
+gem "will_paginate"
+
+gem "bootstrap-will_paginate"
+
+gem 'will_paginate-bootstrap4'
+
 # Bundle edge Rails instead: gem "rails", github: "rails/rails", branch: "main"
 gem "rails", "~> 7.0.7", ">= 7.0.7.2"
```

Rails などのバージョンアップに特定の作業が必要なものはバージョンを固定します

```suggestion
gem "rails", "7.0.7.2"
```

---

rails の指定は上の方に持っていきたい。